cppcache/src/statistics/StatSamplerStats.cpp

```diff
@@ -24,7 +24,7 @@ namespace geode {
 namespace statistics {
 
 StatSamplerStats::StatSamplerStats(StatisticsFactory* statFactory) {
-  statDescriptorArr = new StatisticDescriptor*[2];
+  std::vector<StatisticDescriptor*> statDescriptorArr(2);
```

Comment:
How hard would it be to make this `std::vector<std::shared_ptr<StatisticDescriptor>>` or `unique_ptr` if that is more appropriate?

Reply:
Let's try...